test/run_envoy_bazel_coverage.sh

```diff
@@ -48,31 +48,48 @@ else
   COVERAGE_TARGETS=(//test/...)
 fi
 
+BAZEL_COVERAGE_OPTIONS=()
+
 if [[ "${FUZZ_COVERAGE}" == "true" ]]; then
-  # Filter targets to just fuzz tests.
-  _targets=$(bazel query "attr('tags', 'fuzz_target', ${COVERAGE_TARGETS[*]})")
-  COVERAGE_TARGETS=()
-  while read -r line; do COVERAGE_TARGETS+=("$line"); done \
-      <<< "$_targets"
-  BAZEL_BUILD_OPTIONS+=(
-      "--config=fuzz-coverage"
-      "--test_tag_filters=-nocoverage")
+    # Filter targets to just fuzz tests.
+    _targets=$(bazel query "attr('tags', 'fuzz_target', ${COVERAGE_TARGETS[*]})")
+    COVERAGE_TARGETS=()
+    while read -r line; do COVERAGE_TARGETS+=("$line"); done \
+        <<< "$_targets"
+    BAZEL_COVERAGE_OPTIONS+=(
+        "--config=fuzz-coverage"
+        "--test_tag_filters=-nocoverage")
 else
-  BAZEL_BUILD_OPTIONS+=(
-      "--config=test-coverage"
-      "--test_tag_filters=-nocoverage,-fuzz_target")
+    BAZEL_COVERAGE_OPTIONS+=(
+        "--config=test-coverage"
+        "--test_tag_filters=-nocoverage,-fuzz_target")
 fi
 
 # Output unusually long logs due to trace logging.
-BAZEL_BUILD_OPTIONS+=("--experimental_ui_max_stdouterr_bytes=80000000")
+BAZEL_COVERAGE_OPTIONS+=("--experimental_ui_max_stdouterr_bytes=80000000")
+
+echo "Running bazel coverage with:"
+echo "  Options: ${BAZEL_BUILD_OPTIONS[*]} ${BAZEL_COVERAGE_OPTIONS[*]}"
+echo "  Targets: ${COVERAGE_TARGETS[*]}"
+bazel coverage "${BAZEL_BUILD_OPTIONS[@]}" "${BAZEL_COVERAGE_OPTIONS[@]}" "${COVERAGE_TARGETS[@]}"
 
-bazel coverage "${BAZEL_BUILD_OPTIONS[@]}" "${COVERAGE_TARGETS[@]}"
+echo "Collecting profile and testlogs"
+if [[ -n "${ENVOY_BUILD_PROFILE}" ]]; then
+    cp -f "$(bazel info output_base)/command.profile.gz" "${ENVOY_BUILD_PROFILE}/coverage.profile.gz" || true
+fi
 
-# Collecting profile and testlogs
-[[ -z "${ENVOY_BUILD_PROFILE}" ]] || cp -f "$(bazel info output_base)/command.profile.gz" "${ENVOY_BUILD_PROFILE}/coverage.profile.gz" || true
-[[ -z "${ENVOY_BUILD_DIR}" ]] || find bazel-testlogs/ -name test.log | tar zcf "${ENVOY_BUILD_DIR}/testlogs.tar.gz" -T -
+if [[ -n "${ENVOY_BUILD_DIR}" ]]; then
+    find bazel-testlogs/ -name test.log \
+        | tar cf - -T - \
+        | bazel run "${BAZEL_BUILD_OPTIONS[@]}" //tools/zstd -- \
+                - -T0 --fast -o "${ENVOY_BUILD_DIR}/testlogs.tar.zst"
+    echo "Profile/testlogs collected: ${ENVOY_BUILD_DIR}/testlogs.tar.zst"
+fi
 
-COVERAGE_DIR="${SRCDIR}"/generated/coverage && [[ ${FUZZ_COVERAGE} == "true" ]] && COVERAGE_DIR="${SRCDIR}"/generated/fuzz_coverage
+COVERAGE_DIR="${SRCDIR}/generated/coverage"
+if [[ ${FUZZ_COVERAGE} == "true" ]]; then
+    COVERAGE_DIR="${SRCDIR}"/generated/fuzz_coverage
+fi
```

Comment:
Prefer not mixing in general cleanups, makes the diff hard to read.

Reply:
understood - i was kinda balancing that with the fact that currently this file is hard to read (in my editor) due to ~wierd string formatting